docs/naps/6-contributable-menus.md

```diff
@@ -0,0 +1,517 @@
+contributions could take the same input and create new layers in
+the viewer. By providing dedicated contributions for such actions,
+`napari` can enforce rules about layer editing and layer 
+generation more strictly than for widget contributions,
+which, if taking the `Viewer`, can perform arbitrary actions upon
+all objects within the `Viewer`.
+
+We therefore propose a menu structure that would be easily
+extensible with these new contribution types and provide intuitive
+locations for both plugin developers to add their functionality,
+and users to find it. 
+
+## Detailed Description
+
+<!-- This section should provide a detailed description of the proposed change. It
+should include examples of how the new functionality would be used, intended
+use-cases, and pseudocode illustrating its use. -->
+
+We propose an initial set of contributable menus organized by the napari object
+being acted upon by the actions within the menu, and the likely output of those 
+actions. 
+
+### The `Layers` Menu
+Currently the foremost example of such an object is the napari `Layers`, and this
+menu therefore contains five submenus organized by the types of processing
+the user may wish to perform on the selected `Layer` or `Layers`.
+
+The `Layers` submenus are organized to give the user an immediate
+feeling of what might happen to their `Layers` as a result of clicking
+one of these menu items.
+
+1. `Visualization` - Items in this submenu allow you to generate visualizations from selected layer or layers.
+They do not change the layer data.
+2. `Measure` - Items in this submenu provide utilities for summarising information about your layer's data.
+3. `Edit` - The items in this submenu change the data of your layer through `Filters` or `Transformations`. 
+Additionally `Annotation Tools` provide a location for convenience layer editing tools e.g. `Labels` split/merge actions.
+Items in this submenu **should not** generate new layers, but rather act upon the existing layer data.
+3. `Generate` - Items in this submenu are the main *analysis* actions you can take on your layer. 
+These items should add new layers to the viewer based on analyses and processing of data in your selected layer(s). 
+The five proposed submenus are `Projection`, `Segmentation`, `Classification`, `Registration` and `Tracks`.
+
+Many of the actions in this menu exist in the right click layer context menu. These items
+should be replicated in the `Layers` menu as needed, both to aid discoverability and
+to ensure users are not met with empty menus on initially opening `napari`. It's
+also possible that exceedingly common operations e.g. thresholds, are provided in the future
+by `napari` itself.
+
+### The `Acquisition` Menu
+
+In addition to the `Layers` menu, we add `Acquisition` as a top level menu.
+
+`Acquisition` will contain widgets and utilities for interfacing with microscopes and
+other types of cameras.
```

Comment:
Shoudl this be inside `I/O` rather than on its own? It feels a bit specific for a top level menu at the same level of as `Layers`.

Reply:
disagree, these plugins are in a class of their own, doesn't feel like it fits in IO - I'm fine with top level